### PR TITLE
feat: add 404

### PR DESCRIPTION
**Pull request by jessc93**
#### List of Changes

Add 404 page
#### Motivation and Context

Improve the user experience 

#### Screenshots (if appropriate):
Screenshot 2023-07-07 at 17 30 14

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by datalek** [comment]
Can you add the screenshot?

File: apps/nextjs-website/src/pages/404.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+> = (): GetStaticPropsResult<LayoutProps> => {
+  return { props: { products: [...getProducts()] } };
+};
+
+export default function PageNotFound({ products }: PageNotFoundProps) {
+  const { pageNotFound } = translations;
+
+  return (
+    <Layout products={products}>
+      <SectionTitle
```
Comment (marcobottaro) [inline apps/nextjs-website/src/pages/404.tsx:20]: Forse userei Abstract in questo caso perché mi sembra un po' troppo schiacciata:
image

File: apps/nextjs-website/src/pages/404.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+export type PageNotFoundProps = LayoutProps;
+
+export const getStaticProps: GetStaticProps<
+  PageNotFoundProps
+> = (): GetStaticPropsResult<LayoutProps> => {
+  return { props: { products: [...getProducts()] } };
+};
+
+export default function PageNotFound({ products }: PageNotFoundProps) {
+  const { pageNotFound } = translations;
```
Comment (marcobottaro) [inline apps/nextjs-website/src/pages/404.tsx:16]: Non so se avrei fatto arrivare tutti i testi dalle traduzioni, ma in questo caso ci può anche stare. Forse non farei così le pagine "Privacy" e "Tos"

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/pages/404.tsx (hunk 2 above)
Comment (jessc93) [reply apps/nextjs-website/src/pages/404.tsx:16]: I did not like having hard-coded parts in the code

**Review by kin0992 — APPROVED** [review]

**Review by kin0992 — APPROVED** [review]

**Pull request merged**